### PR TITLE
Edit testCustomAndUpdatedValues with checks for missing kafka configuration

### PR DESCRIPTION
**Pull request by lkral13**
### Type of change
- Refactoring

### Description
- Add method for checking kafka configuration
- Add methods for resolving kafka configuration
- Edit test checks for missing kafka env

### Checklist
- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

**Comment by lkral13** [comment]
@strimzi-ci run tests profile=regression testcase=KafkaST#testCustomAndUpdatedValues

**Comment by lkral13** [comment]
@strimzi-ci run tests profile=regression testcase=KafkaST#testCustomAndUpdatedValues

**Comment by lkral13** [comment]
@strimzi-ci run tests profile=regression testcase=KafkaST

File: systemtest/src/test/java/io/strimzi/systemtest/BaseST.java [hunk 1]
```diff
@@ -376,6 +379,58 @@ protected void checkComponentConfiguration(String podNamePrefix, String containe
         }
     }
 
+    private Properties stringToProperties(String str) {
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/BaseST.java]: Move it to some utils class.

File: systemtest/src/test/java/io/strimzi/systemtest/BaseST.java [hunk 2]
```diff
@@ -376,6 +379,58 @@ protected void checkComponentConfiguration(String podNamePrefix, String containe
...
+            if (split.length == 1) {
+                result.put(split[0], "");
+            } else {
+                result.put(split[0], split[1]);
+            }
+        }
+        return result;
+    }
+
+    private Properties configMap2Properties(ConfigMap cm) {
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/BaseST.java]: Move it to some utils class.

File: systemtest/src/test/java/io/strimzi/systemtest/BaseST.java [hunk 3]
```diff
@@ -376,6 +379,58 @@ protected void checkComponentConfiguration(String podNamePrefix, String containe
...
+        return stringToProperties(cm.getData().get("server.config"));
+    }
+
+    private List<String> getLinesWithoutCommentsAndEmptyLines(String config) {
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/BaseST.java]: Move it to some utils class.

File: systemtest/src/test/java/io/strimzi/systemtest/BaseST.java [hunk 4]
```diff
@@ -376,6 +379,58 @@ protected void checkComponentConfiguration(String podNamePrefix, String containe
...
+        return validLines;
+    }
+
+    protected void checkKafkaConfiguration(String podNamePrefix, Map<String, Object> config, String clusterName) {
+        LOGGER.info("Checking kafka configuration");
+        List<Pod> pods = kubeClient().listPodsByPrefixInName(podNamePrefix);
+
+        Properties properties = configMap2Properties(kubeClient().getConfigMap(clusterName + "-kafka-config"));
+
+        config.forEach((key, val) -> {
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/BaseST.java]: I think use here classic for is more readable.

You can loop over config items and some this: `assertThat(properties, hasEntry(configEntry));`

File: systemtest/src/test/java/io/strimzi/systemtest/BaseST.java [hunk 5]
```diff
@@ -376,6 +379,58 @@ protected void checkComponentConfiguration(String podNamePrefix, String containe
...
+            assertThat(properties.keySet().contains(key), is(true));
+            assertThat(properties.getProperty(key), is(val));
+        });
+
+        for (Pod pod: pods) {
+            ExecResult result = cmdKubeClient().execInPod(pod.getMetadata().getName(), "/bin/bash", "-c", "cat /tmp/strimzi.properties");
+            Properties execProperties = stringToProperties(result.out());
+
+            config.forEach((key, val) -> {
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/BaseST.java]: same as above

**Comment by lkral13** [comment]
@strimzi-ci run tests profile=regression testcase=KafkaST#testCustomAndUpdatedValues

**Comment by lkral13** [comment]
@strimzi-ci run tests profile=regression testcase=KafkaST#testCustomAndUpdatedValues test_only

File: systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java [hunk 6]
```diff
@@ -176,4 +178,34 @@ public static void checkCologForUsedVariable(String varName) {
...
+    public static Properties stringToProperties(String str) {
+        Properties result = new Properties();
+        List<String> list = getLinesWithoutCommentsAndEmptyLines(str);
+        for (String line: list) {
+            String[] split = line.split("=");
+            if (split.length == 1) {
+                result.put(split[0], "");
+            } else {
+                result.put(split[0], split[1]);
+            }
```
Comment (ppatierno) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java]: what about?

```java
result.put(split[0], split.length == 1 ? "" : split[1])
```

Comment (lkral13) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java]: Yeah I think that's better :) thank you, already pushed and ready for merge

**Comment by lkral13** [comment]
@scholzj all conflicts are resolved, ready for merge

**Comment by Frawless** [comment]
@ppatierno could you please merge this pr? We have there several PRs for test fixes and this one block one of these

**Comment by ppatierno** [comment]
@Frawless I see the milestone set to 0.17.0. Do we really need to cherrypick this to 0.17.x branch?

**Comment by Frawless** [comment]
Yes, it contain test fix which should be part of 0.17.x branch

**Pull request merged**